server/src-lib/Hasura/GraphQL/Execute.hs

```diff
@@ -274,43 +277,47 @@ getExecPlan
   -> SchemaCacheVer
   -> GQLReqUnparsed
   -> m (Seq.Seq QExecPlan)
-getExecPlan pgExecCtx planCache userInfo sqlGenCtx enableAL sc scVer reqUnparsed = do
-  planM <-
-    liftIO $ EP.getPlan scVer (userRole userInfo) opNameM queryStr planCache
-  let usrVars = userVars userInfo
-  case planM
-    -- plans are only for queries and subscriptions
-        of
+getExecPlan pgExecCtx planCache userInfo@UserInfo{..} sqlGenCtx enableAL sc scVer reqUnparsed@GQLReq{..} = do
+  liftIO (EP.getPlan scVer userRole _grOperationName _grQuery planCache) >>= \case
```

Comment:
That's totally fair if the idiom makes things less clear to you. The only advantage here is you avoid needing to create a new short-lived name (`planM`) which doesn't carry much meaning (the only purpose is to case on it; we know from the name `getExecPlan` that we should expect some kind of "plan" here). It might be best if I revert this